README.md

```diff
@@ -1,2 +1,135 @@
+steps make sure that the syncing of files between the host and server is correct. Click apply, and finally make sure
+that the new connection is used as default (`ros@localhost:3456` in the list of connections should be bold, otherwise
+use the tick above to set it as default).
+
+![Deployment Settings 2](img/deployment_settings_2.png)
+
+Step 9: Finally, go to `File` &rarr; `Settings` &rarr; `Build, Execution & Deployment` &rarr; `Deployment`
+&rarr; `Options` and change to `Yes` under *Preserve original file permissions (SFTP only)*. Apply and close.
+
+![SFTP Settings](img/sftp_settings.png)
+
+Step 10 (for testing): Right click on your project folder, go to *Deployment* and click on *Upload to ros@localhost:
+3456*. You should get a message in green saying that the files have been transferred.
+
+Step 11 (optional): This example can easily be extended to a project with several ROS packages. Assuming a folder
+structure like
+
+```bash
+ros-pycharm-example
+   └──source
+       ├──package1
+       └──package2
+```
+
+you would just need to `Add new mapping` in step 8 such that you have one mapping per package. Additionally, after you
+have built the package(s) the first time, you can go to `File` &rarr; `Settings` &rarr; `Project: ros-pycharm-example`
+&rarr; `Python Interpreter` and choose *Show All*.
+
+![Add dist packages](img/add_dist_packages.png)
+
+Then, select your new interpreter and click on *Show paths for selected interpreter* and then click on the plus.
+
+![Add dist packages 2](img/add_dist_packages_2.png)
+
+Finally, choose to add `/home/ros/ros_ws/devel/lib/python3/dist-packages` as additional remote path. This will enable
+full IDE capabilities even for the package(s) currently under development in the project.
+
+![Add dist packages 3](img/add_dist_packages_3.png)
+
+## Using the example
+
+Once the server is running in the background, you can use
+the [`connect.sh` script](https://github.com/aica-technology/docker-images/blob/master/scripts/connect.sh)
+to attach new terminal windows by specifying the container name and user. You can use `docker container ls` to lookup
+the names of running containers. In the case of this example:
+
+```shell script
+/.../aica-technology/docker-images/scripts/connect.sh domire8-ros-pycharm-example-ssh --user ros
+```
+
+Using three terminals, invoke `roscore` in one, then `rosrun turtlesim turtlesim_node` in a second, and finally
+`catkin_make && source devel/setup.bash && rosrun turtle_example circler` in the third. You should see the turtle begin
+to swim in circles.
+
+You can also simply use one terminal and
+invoke `catkin_make && source devel/setup.bash && roslaunch turtle_example circler.launch` directly.
+
+With the PyCharm environment properly established, you will also be able to modify the `circler` source script from
+within the IDE instead of from the third terminal and still see the effect on the turtlesim window.
```

Comment:
I see you are sneakily specific here and motion only "modify the source" rather than run it directly. Two things I would mention:
- With automatic syncing enabled, the source file is uploaded on save. So you can change the code in circler, press save, and then go to the connected terminal and immediately do `rosrun turtle_example circler`.
- It's not possible to run a configuration without additional steps


I couldn't get any run configurations to work from the IDE right away. I see this is because the runtime interpreter invokes `ssh://ros@localhost:3456/usr/bin/python3 -u /home/ros/.pycharm_helpers/pydev/pydevconsole.py`, in other works directly invoking the python interpreter without necessarily logging in. Unfortunately this skips the startup script that sources ros, and so it cannot import `rospy`. 

However, we can manually add the ros packages back to the path. If you go to Add / Edit Configurations, then choose one of the scripts (e.g. turtle_example/scripts/circler), you can append the ros and ros_ws packages to the python path under the Environment variables field:
```
PYTHONPATH=$PYTHONPATH:/opt/ros/noetic/lib/python3/dist-packages:/home/ros/ros_ws/devel/lib/python3/dist-packages
```

Once you've done this, it's possible to press Run / Debug and watch the little circler go :)